### PR TITLE
Refactor `plugin-manager.test.js`

### PR DESCRIPTION
**Pull request by sleepwithcoffee**
Related issues: #8603 #7171 

This is a follow up of PR https://github.com/serverless/serverless/pull/11853
Under this PR, I refactor `plugin-manager.test.js` to conform with several common code conventions
* remove `.bind(this)` if arrow function is already sufficient
* Use `to.eventually.be.rejectedWith` instead of `to.be.rejectedWith`, more verbose but screams out promise
* Use `to.eventually.be.fulfilled` instead of `to.not.be.rejected`
* Favor `await` over promises, unless it is a one-line return statement (best practice is always using `await` within function brackets)
* Avoid being verbose in most cases

### CODE REVIEW

**Comment by sleepwithcoffee** [comment]
weird failed test. I tried to run in my local

`npx mocha test/unit/lib/configuration/resolve-provider-name.test.js`
```
  test/unit/lib/configuration/resolve-provider-name.test.js
    ✔ should read name from "provider"
    ✔ should read name from "provider.name"
    ✔ should reject missing "provider.name"
    ✔ should reject invalid "provider.name"
    ✔ should reject missing "provider"


  5 passing (13ms)
```

`node --version` (also tested with  node v12, v14 and v16)
```
v16.19.1
```

**Comment by sleepwithcoffee** [comment]
reverted the troubled piece of code

image

File: test/unit/lib/classes/plugin-manager.test.js [hunk 1]
```diff
@@ -432,19 +431,19 @@ describe('PluginManager', () => {
     });
 
     it('should create an empty cliOptions object', () => {
-      expect(pluginManager.cliOptions).to.deep.equal({});
+      expect(pluginManager.cliOptions).to.be.an('object').empty;
```
Comment (medikoo) [inline test/unit/lib/classes/plugin-manager.test.js:434]: Let's leave original form `to.deep.equal({})` is totally acceptable.

Same in below cases

File: test/unit/lib/classes/plugin-manager.test.js [hunk 2]
```diff
@@ -462,7 +461,7 @@ describe('PluginManager', () => {
       const commands = ['foo', 'bar'];
       pluginManager.setCliCommands(commands);
 
-      expect(pluginManager.cliCommands).to.equal(commands);
+      expect(pluginManager.cliCommands).to.deep.equal(commands);
```
Comment (medikoo) [inline test/unit/lib/classes/plugin-manager.test.js:464]: Let's nott change this test.

In general those old tests, if refactored, they should rather be refactored to `runServerless` usage (see https://github.com/serverless/serverless/tree/main/test#unit-tests). We should not invest time into further mainainace of old tests (unless it helps us in some way, as e.g. getting rid of some dependency)

**Review by medikoo — CHANGES_REQUESTED** [review]
@sleepwithcoffee great thanks for the effort, still let's not adjust the old tests like that.

If we're about to do any work of them, they should be refactored to `runServerless` usage, See https://github.com/serverless/serverless/tree/main/test#unit-tests

**Comment by sleepwithcoffee** [comment]
hi @medikoo these are only meant to be syntactical changes, I do not intend to change the way they work (yet).

**Comment by sleepwithcoffee** [comment]
sure, maybe let me scope down this PR a bit. Some of them are still quite meaningful including highlighting async functions that we might have missed out earlier.
Another meaningful change is to remove `sinon-chai` by using built-in alternative in `chai` alone.

**Comment by medikoo** [comment]
> sure, maybe let me scope down this PR a bit. Some of them are still quite meaningful including highlighting async functions that we might have missed out earlier.
Another meaningful change is to remove sinon-chai by using built-in alternative in chai alone.

@sleepwithcoffee you're doing great work here. What's important in case of old tests (those not configured with `runServerlss`), they're scheduled to be removed in favor of `runServerless` counterparts. So it makes not much sense to work on improvements to those, as it'll anyway be deleted.

Best improvement would be to simply replace them with `runServerless`  based variants as we did in many other files already. Such contribution is very highly welcome.

File: test/unit/lib/classes/plugin-manager.test.js [hunk 3]
```diff
@@ -586,9 +585,9 @@ describe('PluginManager', () => {
       const plugin1 = new ServicePluginMock1();
       plugin1.asyncInit = sinon.stub().returns(Promise.resolve());
       pluginManager.plugins = [plugin1];
-      return pluginManager.asyncPluginInit().then(() => {
-        expect(plugin1.asyncInit.calledOnce).to.equal(true);
-      });
+
+      await pluginManager.asyncPluginInit();
+      expect(plugin1.asyncInit.calledOnce).to.be.true;
```
Comment (medikoo) [inline test/unit/lib/classes/plugin-manager.test.js:590]: Let's not introce such stylistic changes (`to.equal(true)` into `to.be.true`), it brings on extra value, while increases maintenance cost

**Comment by sleepwithcoffee** [comment]
I am closing this PR

**Pull request closed without merging**